src/osp/Active/SysVehicle.cpp

```diff
@@ -308,14 +306,22 @@ ActiveEnt SysVehicle::part_instantiate(PrototypePart& part,
                     DependRes<ImageData2D> imageData = package.get<ImageData2D>(texName);
                     texRes = AssetImporter::compile_tex(imageData, package);
                 }
-                textureResources.push_back(&(*texRes));
+                textureResources.push_back(texRes);
             }
 
             // by now, the mesh and texture should both exist
-
-            CompDrawableDebug& bBocks
-                    = m_scene.reg_emplace<CompDrawableDebug>(
-                        currentEnt, &(*meshRes), std::move(textureResources), m_shader.get(), 0x0202EE_rgbf);
+            using adera::shader::PhongShaderInstance;
+            Package& glResources = m_scene.get_application().get_gl_resources();
+            PhongShaderInstance si;
+            si.m_shaderProgram = glResources.get<adera::shader::Phong>("phong_shader");
+            si.m_textures = std::move(textureResources);
+            si.m_lightPosition = Vector3{10.0f, 15.0f, 5.0f};
+            si.m_ambientColor = 0x111111_rgbf;
+            si.m_specularColor = 0x330000_rgbf;
+            m_scene.reg_emplace<PhongShaderInstance>(currentEnt, std::move(si));
```

Comment:
Does emplace take a variadic argument list?

If so, and you add a constructor to PhongShaderInstance, you can just pass the parameters to the constructor to the reg_emplace<> function directly.